src/lib/certmap/sss_certmap_ldap_mapping.c

```diff
@@ -60,7 +71,229 @@ struct template_table template_table[] = {
+    {"sid", sid_rid_attr, empty},
+    {NULL, NULL, NULL}};
+
+int check_attr_name_and_or_number(TALLOC_CTX *mem_ctx, const char *inp,
+                                  char **_attr_name, int32_t *_number)
+{
+    int ret;
+    char *sep;
+    char *end;
+    char *endptr = NULL;
+    char *attr_name = NULL;
+    int32_t number = 0;
+
+    if (inp == NULL) {
+        attr_name = NULL;
+        number = 0;
+        ret = 0;
+        goto done;
+    }
+
+    sep = strchr(inp, '[');
+    if (sep == NULL) {
+        attr_name = talloc_strdup(mem_ctx, inp);
+        if (attr_name == NULL) {
+            return ENOMEM;
+        }
+        number = 0;
+    } else {
+        end = strchr(sep, ']');
+        if (end == NULL || end == (sep + 1) || *(end + 1) != '\0') {
+            return EINVAL;
+        }
+        number = strtoint32(sep+1, &endptr, 10);
+        if (number == 0 || *endptr != ']') {
+            return EINVAL;
+        }
+
+        if (sep == inp) {
+            attr_name = NULL;
+        } else {
+            attr_name = talloc_strndup(mem_ctx, inp, sep - inp);
+            if (attr_name == NULL) {
+                return ENOMEM;
+            }
+        }
+
+    }
+
+    ret = 0;
+
+done:
+
+    if (ret == 0) {
+        if (_attr_name != NULL) {
+            *_attr_name = attr_name;
+        }
+        if (_number != NULL) {
+            *_number = number;
+        }
+    }
+    return ret;
+}
+
+int check_hex_conversion(const char *inp, bool dec_allowed, bool *_dec,
+                         bool *_upper, bool *_colon, bool *_reverse)
+{
+    int ret;
+    char *sep;
+    bool dec = false;
+    bool upper = false;
+    bool colon = false;
+    bool reverse = false;
+    char *c;
+
+    if (inp == NULL) {
+        ret = 0;
+        goto done;
+    }
+
+    sep = strchr(inp, '_');
+    if (sep == NULL) {
+        if (strcasecmp(inp, "hex") != 0) {
+            if (dec_allowed && strcasecmp(inp, "dec") == 0) {
+                dec = true;
+            } else {
+                ret = EINVAL;
+                goto done;
+            }
+        }
+    } else {
+        if (strncasecmp(inp, "hex", (sep - inp -1)) != 0) {
+            if (dec_allowed && strncasecmp(inp, "dec", (sep - inp -1)) == 0) {
+                dec = true;
+            } else {
+                ret = EINVAL;
+                goto done;
+            }
+        }
+    }
+
+    if (sep != NULL) {
+        for (c = sep + 1; *c != '\0'; c++) {
+            switch(*c) {
+            case 'u':
+            case 'U':
+                upper = true;
+                break;
+            case 'c':
+            case 'C':
+                colon = true;
+                break;
+            case 'r':
+            case 'R':
+                reverse = true;
+                break;
+            default:
+                ret = EINVAL;
+                goto done;
+            }
+        }
+    }
+
+    ret = 0;
+done:
+    if (ret == 0) {
+        if (_dec) {
```

Comment:
I'm all for `if (ptr)` syntax but it's inconsistent with other checks below :)

Reply:
The same in `check_digest_conversion()`.

Reply:
fixed

Reply:
`check_digest_conversion()` still uses `if (_dgst)`

Reply:
Thanks, fixed.